tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py

```diff
@@ -0,0 +1,136 @@
+       registered using agent-auth and when asignn a group ussing  API the change is sync with the cluster.
+tier: 0
+modules:
+    - cluster
+components:
+    - manager
+    - agent
+daemons:
+    - wazuh-db
+    - wazuh-clusterd
+os_platform:
+    - linux
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+references:
+    - https://github.com/wazuh/wazuh-qa/issues/2506
+tags:
+    - cluster
+"""
+import os
+import time
+import pytest
+
+from common import register_agent
+from system import (AGENT_NO_GROUPS, AGENT_STATUS_ACTIVE, AGENT_STATUS_DISCONNECTED, ERR_MSG_FAILED_TO_SET_AGENT_GROUP,
+                    ERR_MSG_CLIENT_KEYS_IN_MASTER_NOT_FOUND, check_agent_groups, check_agent_status, restart_cluster,
+                    check_keys_file, delete_group_of_agents, create_new_agent_group)
+from wazuh_testing.tools.system import HostManager
+
+
+# Hosts
+test_infra_managers = ["wazuh-master", "wazuh-worker1", "wazuh-worker2"]
+test_infra_agents = ["wazuh-agent1"]
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                              'provisioning', 'enrollment_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+tmp_path = os.path.join(local_path, 'tmp')
+
+# Variables
+timeout = 10
+test_group = 'group_test'
+
+
+# Tests
+@pytest.mark.parametrize("test_infra_managers",[test_infra_managers])
+@pytest.mark.parametrize("test_infra_agents",[test_infra_agents])
+@pytest.mark.parametrize("host_manager",[host_manager])
+@pytest.mark.parametrize("initial_status", [AGENT_STATUS_ACTIVE, AGENT_STATUS_DISCONNECTED])
+@pytest.mark.parametrize("agent_target", ["wazuh-master", "wazuh-worker1"])
+def test_assign_agent_to_a_group(agent_target, initial_status, clean_environment, test_infra_managers, test_infra_agents,
+                                 host_manager):
+    '''
+    description: Check agent enrollment process and new group assignment works as expected in a cluster environment.
+                 Check that when an agent pointing to a master/worker node is registered, and when
+                 it's assigned to a new group using API the change is sync with the cluster.
+    wazuh_min_version: 4.4.0
+    parameters:
+        - agent_target:
+            type: string
+            brief: name of the host where the agent will register
+        - initial_status:
+            type: string
+            brief: status of the agent when the assign the new group
+        - clean_enviroment:
+            type: fixture
+            brief: Reset the wazuh log files at the start of the test. Remove all registered agents from master.
```

Comment:
There aren't the variables(test_infra_managers, test_infra_agents, host_manager) defined as parameters of the documentation.